File: The Notebook/Graphs/Depth First Search/dfs.py
visited = []

def initializeVisited(numVertex):
    global visited 
    visited = [0 for i in range(0, numVertex)]
    return


def dfs(adjList, source):
    global visited 
    visited[source] = 1
    for i in range(0, len(adjList[source])):
        u = adjList[source][i]
        if visited[u] == 0:
            dfs(u)
    return


def main():
    numVertex = 100
    adjList = [[] for i in range(0, numVertex)]
    initializeVisited(numVertex)
    # ... put edges in the adjlist ...
    x = 10
    dfs(adjList, x)
    return


if __name__ == "__main__":
    main()

File: The Notebook/Graphs/Depth First Search/dfs.cpp
#include <bits/stdc++.h>
#include <sstream>
#define MAXN 1005

using namespace std;

void dfs(vector<vector<int>> &adjlist, vector<int> &visited, int source){
  visited[source] = 1;
  for(int i = 0; i < (int)adjlist[source].size(); i++){
    int u = adjlist[source][i];
    if(visited[u] == 0){
      dfs(adjlist, visited, u);
    }
  }
  return;
}

int main() {
  vector<vector<int>> adjlist(MAXN, vector<int>());
  vector<int> visited(MAXN, 0);
  int numVertex;
  int a, b;
  cin >> numVertex;

  while(numVertex--){
    cin >> a >> b;
    adjlist[a].push_back(b);
    adjlist[b].push_back(a);
  }
  int source = 13;
  dfs(adjlist, visited, source);

  return 0;
}
